### ISSUE TITLE
watcherService causing high CPU and memory usage when opening workspace

### ISSUE THREAD

**Issue by Silas-Yang** [Microsoft/vscode#40593]
- VSCode Version: 1.19.0
- OS Version: Linux x64 4.3.0-kali-amd64

Steps to Reproduce:

1. Open Folder: select my project folder, and the CPU usage is normal;
2. If I choose "Add Folder to Workspace" instead, the CPU usage is still normal; But if I close VSCode and open it again (resume the workspace), the CPU usage becomes 100% and never falls down.

"node_modules" is a large folder in my project, and I set `files.watcherExclude` for `USER SETTINGS`, `WORKSPACE SETTINGS` and `Folder Settings`.

It is like:
```
"files.watcherExclude": {
		"**/.git/objects/**": true,
		"**/.git/subtree-cache/**": true,
		"**/node_modules/**": true,
		"**/dist/**": true
    }
```

This issue only occurs in workspace. The CPU usage is pretty normal when I open it as a folder.

Reproduces without extensions: Yes

Run `code --status`
```
Version:          Code 1.19.0 (816be6780ca8bd0ab80314e11478c48c70d09383, 2017-12-14T09:56:48.842Z)
OS Version:       Linux x64 4.3.0-kali1-amd64)
CPUs:             Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz (4 x 2999)
Memory (System):  7.66GB (1.60GB free)
Load (avg):       1, 1, 1
VM:               0%
Screen Reader:    no

CPU %	Mem MB	Process
    2	   102	code main
    0	    24	   zygote
    5	   196	     window (settings.json - score-manage2 (工作区) - Visual Studio Code)
   99	  1373	       watcherService
    1	    63	       extensionHost
    1	    55	         electron_node jsonServerMain.js 
    1	    94	     shared-process
    1	    86	   gpu-process

Workspace Stats: 
|  Window (settings.json - score-manage2 (工作区) - Visual Studio Code)
|    Folder (score-manage-frontend): 660 files
|      File types: variables(154) overrides(146) js(127) css(93) less(47)
|                  vue(17) map(12) ttf(11) woff(9) svg(9)
|      Conf files: package.json(2) gulp.js(2) settings.json(1)
|                  webpack.config.js(1)
```

Thanks.

**Comment by bpasero** [comment]
@Silas-Yang could it be possible that you have circular symlinks in your workspace?

/cc @Tyriar

**Comment by Silas-Yang** [comment]
I tried deleting all symlinks in my folder, and the issue is still there. 

1. The issue disappears only when I delete node_module or choose "open folder" instead of "Add folder to workspace". 

2. The problem will occur even if there is only one root folder in workspace.

3.  it won't occur at the first time I add the folder to workspace. If I close VScode and open it to continue with the workspace, the issue occurs.

**Comment by bpasero** [comment]
Does it reproduce if you only open that one folder and set `"files.useExperimentalFileWatcher": true` after a restart?

**Comment by Silas-Yang** [comment]
@bpasero Yes, it does.

I found the reason. It's because I used `cnpm` to install the modules. The simplest workaround may be 
 to use `cnpm install --by=npm` to avoid symlinks.

It seems occurring in Windows too: https://github.com/Microsoft/vscode/issues/35659

But I don't know why `"files.watcherExclude"` doesn't work.
I set it like this: 
```
    "files.useExperimentalFileWatcher": true,
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/node_modules/**": true
    },
```
Thanks.
